File: app/src/main/java/com/robotnec/budget/core/persistence/dao/impl/AccountDaoImpl.java
package com.robotnec.budget.core.persistence.dao.impl;

import com.robotnec.budget.core.persistence.BudgetDatabase;
import com.robotnec.budget.core.persistence.schema.AccountRecord;
import com.robotnec.budget.app.util.Mapper;
import com.robotnec.budget.core.persistence.dao.AccountDao;
import com.robotnec.budget.core.domain.Account;
import com.yahoo.squidb.data.SquidCursor;
import com.yahoo.squidb.sql.Property;

import java.util.List;

/**
 * @author zak <zak@swingpulse.com>
 */
public class AccountDaoImpl extends SquidbDaoTemplate<Account, AccountRecord> implements AccountDao {

    public AccountDaoImpl(BudgetDatabase database) {
        super(database);
    }

    @Override
    AccountRecord fromCursor(SquidCursor cursor) {
        AccountRecord record = new AccountRecord();
        record.readPropertiesFromCursor(cursor);
        return record;
    }

    @Override
    Class<AccountRecord> getRecordClass() {
        return AccountRecord.class;
    }

    @Override
    List<Account> map(List<AccountRecord> items) {
        return Mapper.fromAccountRecords(items);
    }

    @Override
    AccountRecord map(Account item) {
        return Mapper.toRecord(item);
    }

    @Override
    Account map(AccountRecord record) {
        return Mapper.fromRecord(record);
    }

    @Override
    Property.LongProperty getIdProperty() {
        return AccountRecord.ID;
    }
}

File: app/src/main/java/com/robotnec/budget/core/persistence/dao/impl/AccountDaoImpl.kt
package com.robotnec.budget.core.persistence.dao.impl

import com.robotnec.budget.app.util.Mapper
import com.robotnec.budget.core.domain.Account
import com.robotnec.budget.core.persistence.BudgetDatabase
import com.robotnec.budget.core.persistence.dao.AccountDao
import com.robotnec.budget.core.persistence.schema.AccountRecord
import com.yahoo.squidb.data.SquidCursor
import com.yahoo.squidb.sql.Property

/**
 * @author zak zak@swingpulse.com>
 */
class AccountDaoImpl(database: BudgetDatabase) : SquidbDaoTemplate<Account, AccountRecord>(database), AccountDao {

    override val recordClass: Class<AccountRecord>
        get() = AccountRecord::class.java

    override val idProperty: Property.LongProperty
        get() = AccountRecord.ID

    override fun fromCursor(cursor: SquidCursor<AccountRecord>): AccountRecord {
        val record = AccountRecord()
        record.readPropertiesFromCursor(cursor)
        return record
    }

    override fun map(tableModels: List<AccountRecord>): List<Account> {
        return Mapper.fromAccountRecords(tableModels)
    }

    override fun map(item: Account): AccountRecord {
        return Mapper.toRecord(item)
    }

    override fun map(record: AccountRecord): Account {
        return Mapper.fromRecord(record)
    }
}
